### ISSUE TITLE
When making decision on a 'To Investigate' test, API returns `500` with `[could not execute statement; SQL [n/a]; constraint [statistics_check]`

### ISSUE THREAD

**Issue by cailyoung** [reportportal/reportportal#2202]
**Describe the bug**
Recently we noticed that when making decisions on to-investigate tests that the model update would succeed but the test would not be marked as the right type of issue.

Digging through the errors, we saw:

Request (pseudocode):
```
PUT /api/v1/redacted/item/ 

Body:
{"issues":[{"id":767200953,"testItemId":767200953,"issue":{"issueType":"pb001","autoAnalyzed":false,"ignoreAnalyzer":false,"externalSystemIssues":[{"ticketId":"19560","btsUrl":"https://redacted","btsProject":"redacted","url":"redacted","pluginName":"GitHub"}]}}]}
```

API response:
```
{
    "errorCode": 5000,
    "message": "Unclassified error [could not execute statement; SQL [n/a]; constraint [statistics_check]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement]"
}
```

Backend API pod log extract:
```
Caused by: org.postgresql.util.PSQLException: ERROR: new row for relation \"statistics\" violates check constraint \"statistics_check\"
  Detail: Failing row contains (6346533839, -1, 300286, null, 12).
  Where: SQL statement \"UPDATE statistics SET s_counter = s_counter - 1
    WHERE statistics_field_id = defect_field_old_id AND launch_id = cur_launch_id\"
PL/pgSQL function update_defect_statistics() line 105 at SQL statement
```

I checked the function, and it has this code:
```pgsql
    /* decrease launch defects statistics for concrete field */
    UPDATE statistics SET s_counter = s_counter - 1
    WHERE statistics_field_id = defect_field_old_id AND launch_id = cur_launch_id;
```

The failing row looks like this before the change:

```csv
s_id,s_counter,launch_id,item_id,statistics_field_id
6346533839,0,300286,,12
```
...

**Expected behavior**
Making decisions should always be applied regardless of the data state in the DB. We haven't manually edited the DB data at all (I have a readonly user to check the data for situations like this)
...

**Versions:**
 - OS, Browser
 - macOS 14.2.1, Safari 17.2.1
 - Version of RP [find it under Login form, copy as is]
`Current version:API Service: 5.10.1;Index Service: 5.10.0;Jobs Service: 5.10.0;Authorization Service: 5.10.0;Service UI: 5.10.0;`

**Comment by matt-richardson** [comment]
Hi @Kozovskaya :wave:
Is there any update on this one? We're hitting this error again.

If it's likely that it wont get priority, we're happy to contribute a quick fix that fixes to the function, but there might be a larger underlying issue that would be larger than we'd be able to tackle.

**Comment by Kozovskaya** [comment]
@pbortnik Please, could you have a look?

**Comment by matt-richardson** [comment]
Hi @Kozovskaya @pbortnik - can you share an update please?

As I mentioned above, we're happy to contribute a quick fix if that helps?

**Comment by pbortnik** [comment]
Hi, @cailyoung, @matt-richardson 
The main issue seems to be that an item without any statistics is showing up in the database, and when we 'make decision' on empty stats item , the stats try to shift from 0 to -1. This problem is difficult to reproduce, and we would appreciate it greatly if you could figure out and share the exact steps that lead to it. Anyway we suspect the issue might be due to the interaction of multiple threads and database triggers. We are also thinking about removing the db triggers on statistics and implementing their logic directly in a different way.

**Comment by matt-richardson** [comment]
> This problem is difficult to reproduce, and we would appreciate it greatly if you could figure out and share the exact steps that lead to it

Agreed - it happens rarely, so it's not a problem we can reproduce. Sometimes, a record gets in that state somehow - we just dont know why.

**Comment by pbortnik** [comment]
@matt-richardson 
Do you still have that launch with item in the state of bad statistics? If yes, could you please check, if that item is included in launch statistics?

**Comment by matt-richardson** [comment]
> @matt-richardson Do you still have that launch with item in the state of bad statistics? If yes, could you please check, if that item is included in launch statistics?

Sorry, this was months ago - it was cleaned up a long time ago.

**Comment by yurabrovka** [comment]
hi, we have the same issue, have you resolved that?

**Comment by maybeagarden** [comment]
we've run into this too. the times i have seen it is when tests with retries have the retries listed as separate tests rather than as retries against the same test. the same thing that has been reported [here](https://github.com/reportportal/agent-js-playwright/issues/134)

when this happens the totals reported in the launches page and launch widgets are higher than they should be, which makes sense when it see the retry as a separate test

unfortunately i do not know how to reproduce the issue but have seen it numerous times

**Comment by VDkivvi** [comment]
Hi, we have the same issue for some test items in 2026.
